include/tscore/ink_thread.h

```diff
@@ -30,7 +30,15 @@
 
 #include "tscore/ink_hrtime.h"
 #include "tscore/ink_defs.h"
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
 #include <sched.h>
+#if TS_USE_HWLOC
+#include <hwloc.h>
+#include <hwloc/glibc-sched.h>
```

Comment:
I am unable to reproduce the compilation error that occurs when including the header file <hwloc/glibc-sched.h> on freebsd. The error message is: "Please make sure to include sched.h before including glibc-sched.h, and define _GNU_SOURCE before any inclusion of sched.h". I added the undesirable definition of the _GNU_SOURCE macro in the code, as shown below, to ensure that it is defined.

Reply:
hmm, I'm not super familiar with freebsd, but I think this should be added to CXXFLAGS as part of the build setup, not in the source code.  Is Freebsd able to find the numa library?  This feature could be disabled for that OS for now.

Reply:
The _GNU_SOURCE macro is defined in the CMakeLists.txt file as follows:
list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE -DATS_BUILD)
I added the macro in the source code only for testing purposes to ensure that it is indeed defined.
It appears that the error I am encountering is not related to the numa.h library but rather to hwloc.
Thank you for your suggestion. If you have any further insights or recommendations, I would appreciate it.